pkg/bootstrap/envoy_bootstrap.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bootstrap
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"strings"
+	"text/template"
+
+	meshconfig "istio.io/api/mesh/v1alpha1"
+)
+
+// Generate the envoy v2 bootstrap configuration, using template.
+const (
+	// EpochFileTemplate is a template for the root config JSON
+	EpochFileTemplate = "envoy-rev%d.yaml"
+	DefaultCfgDir     = "/var/lib/istio/envoy/envoy_bootstrap_tmpl.json"
```

Comment:
This seems like it should be a parameter, at least to enable testing. Maybe embed the template into a code file to simplify deployment?